url71665d7d68_branscha_lib_scripty_tgz-pJ8-com_sdicons_scripty_ScriptyAutoReplJ8/src/com/sdicons/scripty/cmdlib/FileLibrary.java

```diff
@@ -0,0 +1,301 @@
+            else lFile = resolveFile((String) aFileRepr);
+
+            if (!lFile.isDirectory())
+                throw new CommandException(String.format("Cannot show '%s', it is not a directory.", lFile));
+
+            // Calculate the grep pattern.
+            Pattern lPattern = Pattern.compile(aGrep);
+            // The worklist handles recursion.
+            List<File> lWorkList = new LinkedList<File>();
+            List<File> lResultList = new LinkedList<File>();
+            // Push our starting point on the directory worklist.
+            lWorkList.add(lFile);
+
+            while (lWorkList.size() > 0)
+            {
+                // Get one from the worklist.
+                File lDir = lWorkList.remove(0);
+                File[] lXfiles = lDir.listFiles();
+
+                // A guard, listing files can result in
+                // a null value ... meaning that there probably are
+                // security constraints on the directory.
+                if (lXfiles != null)
+                {
+                    for (File lXFile : lXfiles)
+                    {
+                        if (!".".equals(lXFile.getName()) && !"..".equals(lXFile.getName()))
+                        {
+                            if (lXFile.isDirectory())
+                            {
+                                // Remember if we want dirs.
+                                if (aDirs && lPattern.matcher(lXFile.getCanonicalPath()).matches())
+                                    lResultList.add(lXFile);
+                                // Add to the worklist if recursive.
+                                if (aRecursive) lWorkList.add(lXFile);
+                            } else if (lXFile.isFile())
+                            {
+                                // Remember if we want files.
+                                if (aFiles && lPattern.matcher(lXFile.getCanonicalPath()).matches())
+                                    lResultList.add(lXFile);
+                            }
+                        }
+                    }
+                }
+            }
+
+            for (File lEntry : lResultList)
+            {
+                if (aLambda != null) aEval.eval(lambdaMacro(aLambda, lFile), aCtx);
+
+                if (!aQuiet)
+                {
+                    // Get basic file info.
+                    String lDirFlag = lEntry.isDirectory() ? "d" : " ";
+                    String lReadFlag = lEntry.canRead() ? "r" : " ";
+                    String lWriteFlag = lEntry.canWrite() ? "w" : " ";
+                    String lHiddenFlag = lEntry.isHidden() ? "h" : " ";
+                    long lLastModified = lEntry.lastModified();
+                    // Print the info.
+                    if (aWriter != null)
+                        aWriter.println(String.format("%s%s%s%s  %5$tF %5$tR  %6$s", lDirFlag, lReadFlag, lWriteFlag, lHiddenFlag, lLastModified, lEntry.getCanonicalFile()));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format long: lLastModified (declared at line 192). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 195. Consider formatting this data with &#39;%d&#39; instead.